server/src/main/java/org/opensearch/search/aggregations/StarTreePreComputeCollector.java

```diff
@@ -29,4 +30,18 @@ StarTreeBucketCollector getStarTreeBucketCollector(
         CompositeIndexFieldInfo starTree,
         StarTreeBucketCollector parentCollector
     ) throws IOException;
+
+    /**
+     * Returns the list of dimensions involved in this aggregation, which are required for
+     * merging dimension filters during StarTree precomputation. This is specifically needed
+     * for bucket aggregations to ensure that the correct dimensions are considered when
+     * constructing or merging filters during StarTree traversal.
+     * For metric aggregations, there is no need to specify dimensions since they operate
+     * purely on values within the buckets formed by parent bucket aggregations.
+     *
+     * @return List of dimension field names involved in the aggregation.
+     */
+    default List<String> getDimensionFilters() {
```

Comment:
Can this decision that needs to be taken at a segment level instead of the search request level ?